### PR TITLE
feat: Make EVM wallet connector

### LINKED ISSUE #10284: Make EVM wallet connector
### Is your feature request related to a problem?

I would like to connect my EVM wallets:
- Metamask
- Talisman
- Coinbase wallet

To the Koda. 

### Describe the solution you would like

I would like to see a tabs DOT / EVM where default will be determined by active chain

You can obtain it by `chainProperties.vm` https://github.com/kodadot/nft-gallery/blob/541d7f500e31e025294cbe4aeed57fd5643059c3/libs/static/src/types.ts#L32

Screenshot 2024-05-13 at 12 40 09

## Impl

amazing @brano-hozza hacked EVM login last weekend so code can be reused

https://github.com/Retiazky/shuffle-dao

### Describe alternatives you have considered

We have considered Web3Modal - https://github.com/kodadot/nft-gallery/issues/10137
But that does not work 

### Screenshots

_No response_

### Are there opened related issues?

_No response_

### LINKED ISSUE #10380: Temp fix: Hide elements in sidebar for EVM
### Is there an existing issue?

- [X] I have searched the existing issues

### What happened?

As a normal user I would be waiting for balances to load.

Screenshot 2024-06-03 at 13 18 16

Unfortunatelly it is not possible.

### Please reproduce in steps

1. - #10354 
2. log in
3. balances not loaded

### Expected Behavior

Currently I would hide:
- balances in the sidebar if vm === 'EVM'
- additional things like transfer, teleport ...

### What browsers are you seeing the problem on?

_No response_

### At which address did you encounter bug?

_No response_

### Are you logged in?

None

### Which wallet you are using?

_No response_

### At which chain did you encounter bug?

_No response_

### Screenshots

Screenshot 2024-06-03 at 13 18 39

### Wallet Address & Transaction hash

~

### Relevant log output

_No response_

### Payment link for reward

https://canary.kodadot.xyz/dot/transfer/?target=<My_Polkadot_Address_check_https://github.com/kodadot/nft-gallery/blob/main/REWARDS.md#creating-your-dot-address>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #10381: Profile does not load on Base
### Is there an existing issue?

- [X] I have searched the existing issues

### What happened?

I have visited my profile on base, but It cannot be loaded 😢.

### Please reproduce in steps

1. - #10354
2. Log In 
3. Go to your profile

### Expected Behavior

I expect that my profile would be loaded

### What browsers are you seeing the problem on?

_No response_

### At which address did you encounter bug?

_No response_

### Are you logged in?

None

### Which wallet you are using?

_No response_

### At which chain did you encounter bug?

_No response_

### Screenshots

Screenshot 2024-06-03 at 13 44 56

### Wallet Address & Transaction hash

_No response_

### Relevant log output

```shell
`profileTabsCount`


[
  {
    "message": "Variable \"$interactionIn\" got invalid value \"MINTNFT\" at \"interactionIn[3]\"; Value \"MINTNFT\" does not exist in \"Interaction\" enum. Did you mean the enum value \"MINT\"?",
    "extensions": {
      "code": "BAD_USER_INPUT",
      "exception": {
        "stacktrace": [
          "GraphQLError: Variable \"$interactionIn\" got invalid value \"MINTNFT\" at \"interactionIn[3]\"; Value \"MINTNFT\" does not exist in \"Interaction\" enum. Did you mean the enum value \"MINT\"?",
          "    at /squid/node_modules/graphql/execution/values.js:116:15",
          "    at coerceInputValueImpl (/squid/node_modules/graphql/utilities/coerceInputValue.js:131:9)",
          "    at coerceInputValueImpl (/squid/node_modules/graphql/utilities/coerceInputValue.js:54:14)",
          "    at /squid/node_modules/graphql/utilities/coerceInputValue.js:70:14",
          "    at Array.map (<anonymous>)",
          "    at safeArrayFrom (/squid/node_modules/graphql/jsutils/safeArrayFrom.js:36:23)",
          "    at coerceInputValueImpl (/squid/node_modules/graphql/utilities/coerceInputValue.js:68:50)",
          "    at coerceInputValue (/squid/node_modules/graphql/utilities/coerceInputValue.js:37:10)",
          "    at _loop (/squid/node_modules/graphql/execution/values.js:109:69)",
          "    at coerceVariableValues (/squid/node_modules/graphql/execution/values.js:121:16)"
        ]
      }
    }
  }
]
```

### Payment link for reward

https://canary.kodadot.xyz/dot/transfer/?target=<My_Polkadot_Address_check_https://github.com/kodadot/nft-gallery/blob/main/REWARDS.md#creating-your-dot-address>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [x] Feature
- [x] Refactoring

## Context
❗️❗️ Needs `WALLET_CONNECT_PROJECT_ID`  env

- [x] Closes #10284
- [x] Closes #10380
- [x] Closes #10381

## Needs QA check

- @kodadot/qa-guild please review

## Needs Design check

- @exezbcz please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [x] My fix has changed **something** on UI; 

CleanShot 2024-05-28 at 16 32 02

### CODE REVIEW

**Comment by hassnian** [comment]
@vikiival can you add `WALLET_CONNECT_PROJECT_ID` as env variable

thanks

**Comment by prury** [comment]
its working really nice for me, great job @hassnian :rocket:

**Review by vikiival — COMMENTED** [review]
Not bad, but does it mean that I will be logged off anytime I will use the app? Asking because you deleted localstorage, so Q is where is it persited now?

**Comment by hassnian** [comment]
> but does it mean that I will be logged off anytime I will use the app? Asking because you deleted localstorage, so Q is where is it persited now?

still using localstorage , now I'm persisting the `wallet` store [see](https://github.com/kodadot/nft-gallery/blob/968a6c2d2f9cbe60a95c5af53606625b6bd19910/stores/wallet.ts#L89)

File: components/common/ConnectWallet/ConnectWalletModal.vue [hunk 1]
```diff
@@ -4,163 +4,80 @@
...
-const selectedWalletProvider = ref<BaseDotsamaWallet>()
-const forceWalletSelect = ref(false)
-const wallets = ref<BaseDotsamaWallet[]>([])
-const identityStore = useIdentityStore()
-const { urlPrefix } = usePrefix()
+import ConnectEvm from './ConnectEvm.vue'
+import ConnectSubstrate from './ConnectSubstrate.vue'
+import MnemonicNotice from './MnemonicNotice.vue'
+import WalletTabs from './WalletTabs.vue'
+import { type ChainVM, DEFAULT_VM_PREFIX } from '@kodadot1/static'
```
Comment (vikiival) [inline components/common/ConnectWallet/ConnectWalletModal.vue:46]: Isnt this auto imported?

Comment (hassnian) [reply components/common/ConnectWallet/ConnectWalletModal.vue]: indeed, fixed

File: utils/extension.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ export const getAddress = async (address: string) => {
 
 export const getSelectedAccount = (accounts: WalletAccount[]) => {
   try {
-    const selectedAddress = localStorage.getItem('kodaauth')
+    const selectedAddress = useWalletStore().selected?.address
     const account = accounts.find(
       (account) => account.address === selectedAddress,
     )
```
Comment (Jarsen136) [inline utils/extension.ts:33]: Let's remove it as it's never been used.

File: stores/identity.ts [hunk 3]
```diff
@@ -89,7 +88,7 @@ export const useIdentityStore = defineStore('identity', {
         '1': '0',
         '5': '0',
       },
-      address: localStorage.getItem('kodaauth') || '',
```
Comment (Jarsen136) [inline stores/identity.ts:92]: After releasing to prod env or canary env, the current login address would be automatically logged out. Would you like to be compatible with this case?

File: utils/extension.ts (hunk 2 above)
Comment (hassnian) [reply utils/extension.ts]: done

File: stores/identity.ts (hunk 3 above)
Comment (hassnian) [reply stores/identity.ts:92]: done

**Comment by exezbcz** [comment]
i guess i wont test this one out, betting on prury that he tested it out

**Review by Jarsen136 — APPROVED** [review]

**Comment by vikiival** [comment]
> i guess i wont test this one out, betting on prury that he tested it out

Screenshot 2024-06-04 at 12 16 46


I have tested that 😅

**Comment by hassnian** [comment]
@prury added some things 

- #10380
- #10381
- maintain wallets connected after merge [context](https://github.com/kodadot/nft-gallery/pull/10354#discussion_r1623627779)

adding in this in the developer console 
`````
localStorage.setItem('kodaauth', "1yHddTK4XihjK2DJE8g3oKV5d51TKPmnaCdYgPPTWqm6Ed4") 
localStorage.setItem('wallet', "talisman")
localStorage.setItem('walletname', "test")
`````
and then refreshing the page should show the wallet as connected

File: components/profile/ProfileDetail.vue [hunk 4]
```diff
@@ -704,6 +704,10 @@ const fetchTabsCountByNetwork = async (chain: Prefix) => {
 }
 
 useAsyncData('tabs-empty-result', async () => {
+  if (isEvm.value) {
+    return
```
Comment (vikiival) [inline components/profile/ProfileDetail.vue:708]: maybe consola warn ? 
or check 

- #10387

**Comment by Jarsen136** [comment]
> * maintain wallet connected after merge [context](https://github.com/kodadot/nft-gallery/pull/10354#discussion_r1623627779)
> 
> adding in this in the developer console
> 
> ```
> localStorage.setItem('kodaauth', "1yHddTK4XihjK2DJE8g3oKV5d51TKPmnaCdYgPPTWqm6Ed4") 
> localStorage.setItem('wallet', "talisman")
> localStorage.setItem('walletname', "test")
> ```
> 
> and then refreshing the page should show the wallet as connected

keep this line could probably fixed the bug as it's related to old cache data.
image

**Comment by prury** [comment]
will do another round after comments

**Comment by prury** [comment]
when i connect a EVM account, now its no longer showing sidebar options and other stuff: 

image

**Comment by prury** [comment]
works nicely on mobile as well(installed metamask for mobile and created an ethereum account)

can be solved here:
- the removal of the links on the sidebar(transfer, teleport, etc) only works when you first log in, if you navigate trough the page it comes back

- if i log into evm account and change networks, sidebar gets empty

- base and immutable dissapear from chain list after i log in (to reproduce, log with evm acc and click on drops)


could be followup: 

- i recommend deactivating buying/transfer/mint,nft creation on other chains when the user is logged with evm account

- will on ramp work if i use it on a evm account?

File: components/profile/ProfileDetail.vue (hunk 4 above)
Comment (hassnian) [reply components/profile/ProfileDetail.vue]: done

**Comment by hassnian** [comment]
> keep this line could probably fixed the bug as it's related to old cache data

no need to keep that , the entire local storage gets cleared anyway after calling `resetAuth()`

**Comment by hassnian** [comment]
> works nicely on mobile as well(installed metamask for mobile and created an ethereum account)
> 
> can be solved here:
> 
> * the removal of the links on the sidebar(transfer, teleport, etc) only works when you first log in, if you navigate trough the page it comes back

I can't reproduce that , can you share a video ?

> * if i log into evm account and change networks, sidebar gets empty

only balance and bottom links are hidden as part of #10380

if you have NFTs in that account you would see them

CleanShot 2024-06-05 at 09 58 09@2x

> * base and immutable dissapear from chain list after i log in (to reproduce, log with evm acc and click on drops)

agreed with @exezbcz that once a wallet is selected you can only see chains that have the same vm as the connected wallet

> could be followup:
> 
> * i recommend deactivating buying/transfer/mint,nft creation on other chains when the user is logged with evm account

iirc @exezbcz said atm we should just hide other chains 

> * will on ramp work if i use it on a evm account?

transak supports eth on base 

CleanShot 2024-06-05 at 10 25 26@2x

**Comment by exezbcz** [comment]
> not atm because it will try to buy DOT , just checked and transak does supports eth on `base` and `imx`

skip onramp for now

**Comment by exezbcz** [comment]
what is missing to merge?

**Comment by hassnian** [comment]
> what is missing to merge?

issues are completed, wating for @prury's approval [context](https://github.com/kodadot/nft-gallery/pull/10354/#issuecomment-2148888725)

**Comment by hassnian** [comment]
@vikiival merge conflicts solved

**Pull request merged**